exhibitor-core/src/main/java/com/netflix/exhibitor/core/config/azure/AzurePseudoLock.java

```diff
@@ -0,0 +1,76 @@
+package com.netflix.exhibitor.core.config.azure;
+
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.azure.storage.blob.ListBlobItem;
+import com.netflix.exhibitor.core.azure.AzureClient;
+import com.netflix.exhibitor.core.config.PseudoLockBase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AzurePseudoLock extends PseudoLockBase
+{
+    private final AzureClient      client;
+    private final String        container;
+
+    /**
+     * @param client the Azure client
+     * @param container the Azure container
+     * @param lockPrefix uri prefix
+     * @param timeoutMs max age for locks
+     * @param pollingMs how often to poll Azure
+     */
+    public AzurePseudoLock(AzureClient client, String container, String lockPrefix, int timeoutMs, int pollingMs)
+    {
+        super(lockPrefix, timeoutMs, pollingMs);
+        this.client = client;
+        this.container = container;
+    }
+
+    /**
+     * @param client the Azure client
+     * @param container the Azure container
+     * @param lockPrefix uri prefix
+     * @param timeoutMs max age for locks
+     * @param pollingMs how often to poll Azure
+     * @param settlingMs how long to wait for Azure to reach consistency
+     */
+    public AzurePseudoLock(AzureClient client, String container, String lockPrefix, int timeoutMs, int pollingMs, int settlingMs)
```

Comment:
this to me looks awfully similar to the other container: I'd lose one or the other - having both feels like overkill